src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -423,21 +437,52 @@ public void finish() {
             metrics.MEAN_PROBE_COVERAGE   = metrics.ON_PROBE_BASES / (double) metrics.PROBE_TERRITORY;
             metrics.FOLD_ENRICHMENT       = (metrics.ON_PROBE_BASES/ denominator) / ((double) metrics.PROBE_TERRITORY / metrics.GENOME_SIZE);
 
-            calculateTargetCoverageMetrics();
+            // Get Theoretical Het SNP Sensitivity and Calculate Target Coverage Metrics
+            final double[] coverageDistribution = calculateTargetCoverageMetrics();
+
+            // Construct and write the outputs
+            for (int i=0; i<baseQHistogramArray.length; ++i) {
+                baseQHisto.increment(i, baseQHistogramArray[i]);
+            }
+
+            // Construct and write the outputs
+            for (int i = 0; i < coverageDistribution.length; ++i) {
+                depthHisto.increment(i, coverageDistribution[i]);
+            }
+
+            final int sampleSize = 10000;
+            final double logOddsThreshold = 3.0;
+            final double [] depthDoubleArray = TheoreticalSensitivity.normalizeHistogram(depthHisto);
+            final double [] baseQDoubleArray = TheoreticalSensitivity.normalizeHistogram(baseQHisto);
+            metrics.HET_SNP_SENS = TheoreticalSensitivity.hetSNPSensitivity(depthDoubleArray, baseQDoubleArray, sampleSize, logOddsThreshold);
+            metrics.HET_SNP_Q = QualityUtil.getPhredScoreFromErrorProbability((1 - metrics.HET_SNP_SENS));
+
             calculateGcMetrics();
         }
 
         /** Calculates how much additional sequencing is needed to raise 80% of bases to the mean for the lane. */
-        private void calculateTargetCoverageMetrics() {
+        private double[] calculateTargetCoverageMetrics() {
             final int[] depths = new int[(int) this.metrics.TARGET_TERRITORY];  // may not use entire array
+            final double[] coverageDistribution = new double[coverageCap+1];
             int zeroCoverageTargets = 0;
             int depthIndex = 0;
             double totalCoverage = 0;
             int basesConsidered = 0;
 
             for (final Coverage c : this.coverageByTarget.values()) {
+                //add to coverage distribution for het sensitivity regardless of low target coverage
                 if (!c.hasCoverage()) {
                     ++zeroCoverageTargets;
+                    final int[] targetDepths = c.getDepths();
```

Comment:
I'm confused here, if `c` doesn't have coverage, we know that `c.getDepths()` is an array of zeroes, so every depth is going to satisfy `depth < coverageCap`.  So it seems like the following code would have the same effect:
```
if (!c.hasCoverage()) {
    ++zeroCoverageTargets;
    coverageDistribution[0] += targetDepths.getDepths().length
}
```